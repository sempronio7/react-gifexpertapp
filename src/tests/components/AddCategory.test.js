import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe ('Pruebas en <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    })
    
    test('debe mostrarse correctamente', ()=>{
        expect (wrapper).toMatchSnapshot();
    })

    test('debe cambiar la caja de texto', ()=>{
        const input = wrapper.find('input');
        const value = 'hola';
        input.simulate('change', {
            target: {
                value: value
            }
        });

        expect(wrapper.find('p').text().trim()).toBe(value);
    })

    test('no debe postear la informacion onSubmit', ()=>{
        wrapper.find('form').simulate('submit', {preventDefault: () => {}});
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('debe llamar el setCategories y limpiar la caja de texto', ()=>{
        const value = 'hola mundo';

        //1. simular el inputChange
        wrapper.find('input').simulate('change', {target: {value}});

        //2. simular el submit
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        //3. debe ser llamado el setCategories
        expect(setCategories).toHaveBeenCalled();
        
        //4. el valor de input debe estar ''
        expect(wrapper.find('input').prop('value')).toBe('');
        

    })

})