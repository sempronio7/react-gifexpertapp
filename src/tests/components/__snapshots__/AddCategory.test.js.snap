// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> debe mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
